app/core/store/modules/prepaids.js

```diff
@@ -269,66 +270,74 @@ export default {
       await Promise.all(requests)
     },
 
-    async revokeLicenses (_, { members, sharedClassroomId }) {
-      const students = members.map(data => new User(data)).filter(u => u.isEnrolled())
+    async revokeLicenses (_, { members, sharedClassroomId, confirmed = false, updateUserProducts = false }) {
+      const students = members
+        .map(data => data instanceof User ? data : new User(data))
+        .filter(u => u.isEnrolled() && u.prepaidType() === 'course')
 
       const existsLicenseToRevoke = students.length > 0
       if (!existsLicenseToRevoke) {
         noty({ text: `No licenses applied to selected student(s).` })
         return
       }
 
-      let confirmed = false
-      await new Promise((resolve) => noty({
-        text: `Revoking a license will make it available to apply to other students. Students will no longer be able to access paid content, but their progress will be saved. Please confirm you'd like to proceed.`,
-        buttons: [
-          {
-            addClass: 'btn btn-primary',
-            text: 'Ok',
-            onClick: function ($noty) {
-              confirmed = true
-              $noty.close()
-              resolve()
-            }
-          },
-          {
-            addClass: 'btn btn-danger',
-            text: 'Cancel',
-            onClick: function ($noty) {
-              $noty.close()
-              resolve()
+      if (!confirmed) {
+        await new Promise((resolve) => noty({
+          text: `Revoking a license will make it available to apply to other students. Students will no longer be able to access paid content, but their progress will be saved. Please confirm you'd like to proceed.`,
+          buttons: [
+            {
+              addClass: 'btn btn-primary',
+              text: 'Ok',
+              onClick: function ($noty) {
+                confirmed = true
+                $noty.close()
+                resolve()
+              }
+            },
+            {
+              addClass: 'btn btn-danger',
+              text: 'Cancel',
+              onClick: function ($noty) {
+                $noty.close()
+                resolve()
+              }
             }
-          }
-        ]
-      }))
+          ]
+        }))
 
-      if (!confirmed) {
-        return
+        if (!confirmed) {
+          return
+        }
       }
 
-      const promises = []
       for (const student of students) {
         const courseProducts = student.activeProducts('course')
-        courseProducts.map(product => {
-          let prepaid = new Prepaid({
+        await Bluebird.map(courseProducts, async product => {
```

Comment:
why do we need Bluebird here?

Reply:
@mrfinch  no actual need for it. I just like more the bluebird syntax, because it seems to be more simple for me. One `bluebird.map` instead of `Array.map`, `.push()` and `Promise.all()`. Also no need to maintain a `promises` array.

Reply:
Haha ok. Though, I would prefer not to add a lot of new styles of code for similar things. Also, bluebird package hasn't been updated for 3 years so preferably better not to use it much!!